spec/services/idv/steps/in_person/address_step_spec.rb

```diff
@@ -92,6 +88,25 @@
             state:,
           )
         end
+
+        context 'when initially entering the residential address' do
+          it 'leaves the "same_address_as_id" attr as false' do
+            flow.flow_session[:pii_from_user][:same_address_as_id] = 'false'
+            step.call
+
+            expect(flow.flow_session[:pii_from_user][:same_address_as_id]).to eq('false')
+          end
+        end
+
+        context 'when updating the residential address' do
+          it 'sets the "same_address_as_id" in the flow session to false' do
+            flow.flow_session[:pii_from_user][:same_address_as_id] = 'true'
+            flow.flow_session[:pii_from_user][:address1] = '123 New Residential Ave'
+
+            step.call
+            expect(flow.flow_session[:pii_from_user][:same_address_as_id]).to eq('false')
+          end
+        end
```

Comment:
There's an additional case to cover here.
```suggestion
        context 'when updating the residential address' do
          before(:each) do
            flow.flow_session[:pii_from_user][:address1] = '123 New Residential Ave'
          end
          context 'the user previously indicated that the residential address matched the state ID' do
            before(:each) do
              flow.flow_session[:pii_from_user][:same_address_as_id] = 'true'
            end
            it 'sets the "same_address_as_id" in the flow session to false' do
              step.call
              expect(flow.flow_session[:pii_from_user][:same_address_as_id]).to eq('false')
            end
          end
          context 'the user previously indicated that the residential address did not match the state ID' do
            before(:each) do
              flow.flow_session[:pii_from_user][:same_address_as_id] = 'false'
            end
            it 'leaves the "same_address_as_id" in the flow session as false' do
              step.call
              expect(flow.flow_session[:pii_from_user][:same_address_as_id]).to eq('false')
            end
          end
        end
```

Reply:
Thanks, @NavaTim ! Is there a reason you'd use `before(:each)` for the single tests (example: line 115)?

For me it's a little more intuitive to just use `before` until there is more than one test within the context. If It's a `before(:each)` with only one test, I might spend time trying to look for another test.

I tested out both locally and they both work. But before I pushed it up I thought I'd see if there was a reason or if it's personal preference.

Reply:
[`:each`/`:example` is the default](https://rubydoc.info/gems/rspec-core/RSpec/Core/Hooks%3abefore) (IMO for good reason), including it makes that explicit.